Client/Utils/FaviconFetcher.swift

```diff
@@ -99,30 +98,20 @@ open class FaviconFetcher: NSObject, XMLParserDelegate {
         return deferred
     }
 
-    lazy fileprivate var alamofire: SessionManager = {
-        let configuration = URLSessionConfiguration.default
-        var defaultHeaders = SessionManager.default.session.configuration.httpAdditionalHeaders ?? [:]
-        defaultHeaders["User-Agent"] = FaviconFetcher.userAgent
-        configuration.httpAdditionalHeaders = defaultHeaders
-        configuration.timeoutIntervalForRequest = 5
-        return SessionManager(configuration: configuration)
-    }()
+    lazy fileprivate var urlSession: URLSession = makeUrlSession(userAgent: FaviconFetcher.userAgent, isEphemeral: false, timeout: 5)
 
     fileprivate func fetchDataForURL(_ url: URL) -> Deferred<Maybe<Data>> {
         let deferred = Deferred<Maybe<Data>>()
-        alamofire.request(url).response { response in
-            // Don't cancel requests just because our Manager is deallocated.
-            withExtendedLifetime(self.alamofire) {
-                if response.error == nil {
-                    if let data = response.data {
-                        deferred.fill(Maybe(success: data))
-                        return
-                    }
-                }
-                let errorDescription = (response.error as NSError?)?.description ?? "No content."
-                deferred.fill(Maybe(failure: FaviconFetcherErrorType(description: errorDescription)))
+        urlSession.dataTask(with: url) { (data, _, error) in
```

Comment:
We could also have a `fetchData()` equivalent of the `fetchJSON()` helper for cases like this so that we at least DRY up the filling of the `Deferred`/error handling.